soc/espressif/esp32s2/soc.c

```diff
@@ -143,6 +119,14 @@ void __attribute__((section(".iram1"))) __esp_platform_start(void)
 #endif /* !CONFIG_MCUBOOT */
 
 	esp_intr_initialize();
+
+#if CONFIG_ESP_SPIRAM
+	/* Init Shared Multi Heap for PSRAM */
+	if (esp_psram_smh_init()) {
```

Comment:
I noticed the PSRAM is started at very early kernel stage without even logging or printk, it would not be better to move it to POST_KERNEL with the highest priority instead? It would better debugging experience since tracing and kernel would be available and I suppose the external heap would be consumed for POST_KERNEL components as well like wifi, camera, display or I2S.

Reply:
Provided we are sure the `.ext_ram.bss` section won't be used in PRE_KERNEL, we could do it. However, if its only merit is to inform the user, there are log messages informing user about PSRAM initialization, so I'm not sure it would be much useful.